src/client/views/UpdatePattern.tsx

```diff
@@ -1,11 +1,76 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import noteService from "../services/pattern";
+import { useLocation, useNavigate, useParams } from "react-router-dom";
+import { useForm } from "../utils/use-form";
+import patternService from "../services/pattern";
+import { IPattern } from "../utils/types";
 
 interface UpdatePatternProps {}
 
 const UpdatePattern = (props: UpdatePatternProps) => {
+	const [pattern, setPattern] = useState<IPattern | undefined>();
+	const [content, setContent] = useState<string>();
+	const [title, setTitle] = useState<string>();
+	const { id } = useParams();
+	const { state } = useLocation();
+	const navigate = useNavigate();
+
+	const handleChanges = (
+		e: React.ChangeEvent<HTMLInputElement & HTMLTextAreaElement>
+	) => {
+		setPattern((prev) => ({
+			...prev,
+			[e.target.name]: e.target.value,
+		}));
+	};
+
+	useEffect(() => {
+		if (!state) {
+			if (!id) return;
+			patternService.getOnePattern(id).then((pattern: IPattern) => {
+				setPattern(pattern);
+				setContent(pattern.content);
+			});
+			// .catch(e => Toast.error(e.message));
+		}
```

Comment:
```suggestion
		if (state) return;
		if (!id) return;
		patternService.getOnePattern(id).then((pattern: IPattern) => {
			setPattern(pattern);
			setContent(pattern.content);
		});
		// .catch(e => Toast.error(e.message));
	}
```
You can do an early return for the state check as well to reduce the layers of nesting here.